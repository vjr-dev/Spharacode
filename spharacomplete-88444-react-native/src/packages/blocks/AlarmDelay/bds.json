{
  "bds": {
    "name": "AlarmDelay",
    "id": "AlarmDelay",
    "platform": [
      "desktop",
      "mobile"
    ],
    "author": "CMT",
    "summary": null,
    "license": "BSD",
    "version": "1.0.0",
    "pages": [
      {
        "AlarmDelay": {
          "hotspots": [
          ]
        }
      }
    ]
  }
}
