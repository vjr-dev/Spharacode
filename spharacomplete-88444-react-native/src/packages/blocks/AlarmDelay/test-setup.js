// test-setup.js
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NativeModules } from 'react-native';
configure({ adapter: new Adapter() });

NativeModules.RNCAsyncStorage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    mergeItem: jest.fn(),
    clear: jest.fn(),
    getAllKeys: jest.fn(),
    flushGetRequests: jest.fn(),
    multiGet: jest.fn(),
    multiSet: jest.fn(),
    multiRemove: jest.fn(),
    multiMerge: jest.fn(),
};
NativeModules.RNPermissions = {
    RNPermissions: jest.fn(),
};

jest.mock("react-native-shake", () => ({
   
    // addListener: jest.fn().mockImplementation((cb) => cb()),
  }));

  jest.mock("@react-native-async-storage/async-storage", () => ({
    getItem: jest
      .fn()
      .mockImplementationOnce((key) => {
        console.log("_----key", key);
        if (key === "Token") return "0";
        return "Token";
      })
      .mockImplementation(() => null),
    setItem: jest.fn(),
  }));