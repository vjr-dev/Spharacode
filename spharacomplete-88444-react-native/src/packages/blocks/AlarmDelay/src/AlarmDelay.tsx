// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";

import {
    StyleSheet,
    Text,
    View,
    Image,
    ImageBackground,
    Dimensions,
    SafeAreaView,
    TouchableOpacity,
    Modal,
    ActivityIndicator,
    StatusBar
    
} from "react-native";
import AlarmDelayController, { Props } from "./AlarmDelayController";
import { back1, back2, image_back } from "./assets";
import { COLORS } from "../../../framework/src/Globals";
const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;
import { RFValue } from "react-native-responsive-fontsize";
import Slider from '@react-native-community/slider';  



export default class AlarmDelay extends AlarmDelayController {
    constructor(props: Props) {
        super(props);
    }

    
    

    render() {
        return (
            <SafeAreaView style={styles.container1}>
                <View style={styles.child}>
                    <StatusBar
                        animated={true}
                        backgroundColor="#454545"
                        barStyle={"light-content"}
                        showHideTransition={"slide"}
                    // hidden={true}
                    />
                    <ImageBackground
                        source={back1}
                        style={styles.image}>
                        <ImageBackground
                            source={back2}
                            style={styles.image2}>


                            <View
                                style={styles.view1}
                            >
                                {/* <TouchableOpacity style={styles.button1} onPress={() => this.goback()}>
                                    <Image
                                        source={image_back}
                                        style={styles.backbutton}
                                    />
                                </TouchableOpacity> */}
                                <TouchableOpacity 
                                testID="backBtnId"
                                style={styles.button1} 
                                onPress={() => this.goback()}>
                                    <Image
                                        source={image_back}
                                        style={styles.backbutton}
                                    />
                                </TouchableOpacity>
                            </View>

                            <Text style={{ fontSize: 20, color: COLORS.white, marginTop: RFValue(40), width: windowWidth - 20, }}>Alarm Delay.</Text>
                            <Text style={{ fontSize: 16, color: COLORS.ultralightwhite, marginTop: RFValue(10), width: windowWidth - 20 }}>Set number of seconds from when the SOS button is pushed until the alarm triggerd.</Text>




                            <View style={{ height: RFValue(50), width: windowWidth, flexDirection: "row", alignItems: "center", marginTop: 25, justifyContent: "space-between" }}>
                                <Text style={{ fontSize: 20, color: COLORS.lightwhite, marginLeft: RFValue(10) }}>Voice Activation alarm delay.</Text>
                                <TouchableOpacity
                                testID="setClickId"
                                    onPress={() => this.setclick()}
                                    style={{ height: "100%", alignItems: "center", justifyContent: "center" }}>
                                    <Text style={{ marginRight: RFValue(15), color: COLORS.darkorange, fontSize: 20 }}>SET</Text>
                                </TouchableOpacity>
                            </View>



                            <View style={{ height: RFValue(100), width: windowWidth, alignSelf: 'center', backgroundColor: COLORS.backgroundGray, alignItems: "center", justifyContent: "space-evenly", flexDirection: "row" }}>
                                <Text style={{ color: COLORS.lightwhite, marginLeft: RFValue(30), marginTop: RFValue(16) }}>5</Text>
                                <View style={{ height: "100%", width: RFValue(320), justifyContent: "center" }}>
                                    <View style={{ width: RFValue(100), left: this.state.value1 * RFValue(4) - RFValue(20), backgroundColor: COLORS.orangelight, borderRadius: RFValue(50) }}>
                                        <Text style={{ textAlign: 'center', color: COLORS.black, }}>{this.state.value1} Second</Text>
                                    </View>
                                    <View style={{ width: RFValue(320), alignItems: "center", marginTop: 5 }}>
                                        <Slider
                                        testID="silderValue"
                                            style={{ width: RFValue(275), alignItems: "center", justifyContent: "center" }}
                                            value={this.state.value1}
                                            // onSlidingStart={true}
                                            minimumValue={5}
                                            maximumValue={60}
                                            onValueChange={(V) => this.setState({ value1: V })}
                                            minimumTrackTintColor={COLORS.darkorange}
                                            maximumTrackTintColor={COLORS.ultralightwhite}
                                            thumbTintColor={COLORS.white}
                                            step={1}
                                        /></View>
                                </View>
                                <Text style={{ color: COLORS.lightwhite, marginRight: RFValue(30), marginTop: RFValue(16) }}>60</Text>
                            </View>


                            {/* ------ 2 ------ */}

                            <View style={{ height: RFValue(50), width: windowWidth, flexDirection: "row", alignItems: "center", marginTop: 25, justifyContent: "space-between" }}>
                                <Text style={{ fontSize: 20, color: COLORS.lightwhite, marginLeft: RFValue(10) }}>Gesture trigger delay.</Text>
                                <TouchableOpacity
                                testID="SetClickId"
                                    onPress={() => this.setclick()}
                                    style={{ height: "100%", alignItems: "center", justifyContent: "center" }}>
                                    <Text style={{ marginRight: RFValue(15), color: COLORS.darkorange, fontSize: 20 }}>SET</Text>
                                </TouchableOpacity>
                            </View>



                            <View style={{ height: RFValue(100), width: windowWidth, alignSelf: 'center', backgroundColor: COLORS.backgroundGray, alignItems: "center", justifyContent: "space-evenly", flexDirection: "row" }}>
                                <Text style={{ color: COLORS.lightwhite, marginLeft: RFValue(30), marginTop: RFValue(16) }}>5</Text>
                                <View style={{ height: "100%", width: RFValue(320), justifyContent: "center" }}>
                                    <View style={{ width: RFValue(100), left: this.state.value2 * RFValue(4) - RFValue(20), backgroundColor: COLORS.orangelight, borderRadius: RFValue(50) }}>
                                        <Text style={{ textAlign: 'center', color: COLORS.black, }}>{this.state.value2} Second</Text>
                                    </View>
                                    <View style={{ width: RFValue(320), alignItems: "center", marginTop: 5 }}>
                                        <Slider
                                        testID="sildervalue2"
                                        style={{ width: RFValue(275), alignItems: "center", justifyContent: "center" }}
                                        value={this.state.value2}
                                        // onSlidingStart={true}
                                        minimumValue={5}
                                        maximumValue={60}
                                        onValueChange={(V) => this.setState({ value2: V })}
                                        minimumTrackTintColor={COLORS.darkorange}
                                        maximumTrackTintColor={COLORS.ultralightwhite}
                                        thumbTintColor={COLORS.white}

                                        step={1}
                                    /></View>
                                </View>
                                <Text style={{ color: COLORS.lightwhite, marginRight: RFValue(30), marginTop: RFValue(16) }}>60</Text>
                            </View>



                            <Modal visible={this.state.Loader} transparent={true}>
                                <View style={{ height: windowHeight, width: windowWidth, alignItems: "center", justifyContent: "center" }}>
                                    <View
                                        style={{ height: 100, width: windowWidth - 20, alignSelf: "center", justifyContent: "center", alignItems: "center", }} >
                                        <ActivityIndicator
                                            animating={true}
                                            size={"large"}
                                            color="#f07233"
                                        />
                                    </View>
                                </View>
                            </Modal>

                        </ImageBackground>
                    </ImageBackground>
                </View>
            </SafeAreaView >
            
        );
    }
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        height: windowHeight, width: windowWidth,
        backgroundColor: "#ffffffff"
    },
    image: {
        height: windowHeight, width: windowWidth,
    },
    image2: {
        height: windowHeight, width: windowWidth, alignItems: "center",
    },
    container1: { flex: 1, backgroundColor: "#454545" },
    child: { width: windowWidth, },
    backbutton: { height: 15, width: 15, tintColor: "#f07233" },
    view1: { height: 60, width: windowWidth, justifyContent: "center" },
    button1: {
        width: 15,
        height: 15,
        padding: 25,
        justifyContent: "center",
        alignItems: "center",
    },

});

// Customizable Area End