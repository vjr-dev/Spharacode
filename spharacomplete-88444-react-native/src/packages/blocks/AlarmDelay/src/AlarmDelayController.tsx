// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
    getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";
import Dashboard from '../../DashBord/src/DashbordController';
import AsyncStorage from "@react-native-async-storage/async-storage";



export const configJSON = require("./config");


export interface Props {
    navigation: any;
    id: string;
    
    
}

interface S {
    
    Loader: boolean
    value1: any
    value2: any
    Userdata: any
    Token: any

    
}

interface SS {
    id: any;
    
    
}

export default class AlarmDelay extends BlockComponent<Props, S, SS> {
    
    SetsettingID: any = "";
    private shackInstance: Dashboard;
    

    constructor(props: Props) {
        super(props);
       this.shackInstance = new Dashboard(props);
        this.receive = this.receive.bind(this);

        
        this.subScribedMessages = [
            
            getName(MessageEnum.RestAPIResponceMessage),
            getName(MessageEnum.RestAPIRequestMessage),
            getName(MessageEnum.RestAPIResponceEndPointMessage),
            getName(MessageEnum.RestAPIRequestBodyMessage),
            getName(MessageEnum.RestAPIRequestHeaderMessage),
            getName(MessageEnum.RestAPIRequestMethodMessage),
            
        ];

        this.state = {
            
            Loader: false,
            value1: 10,
            value2: 5,
            Userdata: "",
            Token: "",

            
        };
        runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  
    }

    async componentDidMount() {
        this.setState({ Userdata: this.props.route.params.data })
        this.setState({ Token: await AsyncStorage.getItem("Token") })
        this.setState({ value1: this.state.Userdata.voice_activation_alarm_delay, value2: this.state.Userdata.gesture_delay })
    }
    async receive(from: string, message: Message) {
        
        runEngine.debugLog("Message Recived", message);

        if (message.id === getName(MessageEnum.RestAPIResponceMessage)) //istanbul ignore next
          {
            const apiRequestCallId = message.getData(
                getName(MessageEnum.RestAPIResponceDataMessage)
            );

            let responseJson = message.getData(
                getName(MessageEnum.RestAPIResponceSuccessMessage)
            );

            if (apiRequestCallId === this.SetsettingID) //istanbul ignore next
            {
                if (responseJson != null) //istanbul ignore next
                {
                    
                    if (responseJson.data != null) //istanbul ignore next
                    {

                        this.setState({ Loader: false })
             
                        await AsyncStorage.setItem("ShakeEventValue", JSON.stringify(responseJson?.data?.attributes?.gesture_delay))
                    } else  //istanbul ignore next
                    {
                        this.setState({ Loader: false })
                        alert("ERROR")
                    }


                }
            }
        }
    }

    setclick = async () => {

        this.setState({ Loader: true })
        const header = {
            "Content-Type": configJSON.UpdatesettingAPiContentType,
            token: this.state.Token
        };
        const attrs = {
            voice_activation_alarm_delay: this.state.value1,
            gesture_delay: this.state.value2,
        };
        const data1 = {
            attributes: attrs
        };
        const httpBody = {
            data: data1
        };
        
        const requestMessage = new Message(
            getName(MessageEnum.RestAPIRequestMessage)
        );

        this.SetsettingID = requestMessage.messageId;


        requestMessage.addData(
            getName(MessageEnum.RestAPIResponceEndPointMessage),
            configJSON.UpdatesettingAPiEndPoint
        );
        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestHeaderMessage),
            JSON.stringify(header)
        );

        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestBodyMessage),
            JSON.stringify(httpBody)
        );

        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestMethodMessage),
            configJSON.UpdatesettingAPiMethod
        );

        runEngine.sendMessage(requestMessage.id, requestMessage);

        return true;

    }
    

    goSignupScreen=() => {
        const { navigation }: any = this.props
        navigation.navigate("SignUpScreen")

    }
    goback =() => {
        const { navigation }: any = this.props
        navigation.goBack()
    }
    
}
// Customizable Area End