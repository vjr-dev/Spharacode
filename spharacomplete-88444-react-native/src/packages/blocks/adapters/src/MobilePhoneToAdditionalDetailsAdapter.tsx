import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
