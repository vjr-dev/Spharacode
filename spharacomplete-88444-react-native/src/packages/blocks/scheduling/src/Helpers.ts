/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});