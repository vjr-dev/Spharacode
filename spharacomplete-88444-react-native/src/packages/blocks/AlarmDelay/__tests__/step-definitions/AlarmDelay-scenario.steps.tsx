//@ts-nocheck
//@ts-ignore
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "framework/src/Helpers";
import React from "react";
import AlarmDelay from "../../src/AlarmDelay";
import MessageEnum, { getName } from 'framework/src/Messages/MessageEnum';
import { Message } from 'framework/src/Message';
import { runEngine } from 'framework/src/RunEngine';

const CreateEventRes = {}
const screenProps = {
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    dispatch: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
   
    
  },
  id: "AlarmDelay",
};

const feature = loadFeature(
  "./__tests__/features/AlarmDelay-scenario.feature"
);

jest.mock("react-native-shake", () => ({
  RNShake: jest.fn()
  // addListener: jest.fn().mockImplementation((cb) => cb()),
}));
jest.mock("react-native-sound", () =>({
  setCategory : jest.fn(),
}));

jest.mock("react-native-system-setting", () => ({
  removeVolumeListener: jest.fn(),
  getVolume: jest.fn(() => Promise.resolve("test")),
  addVolumeListener: jest
    .fn()
    .mockImplementation((callback: any) => callback({ value: 2 })),
}));

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  
    jest.doMock("../../../../mobile/sample.mp3", () => ({}));
  
  });

  test("User navigates to AlarmDelay", ({ given, when, then }) => {
    let AlarmDelayWrapper: ShallowWrapper;
    let instance: AlarmDelay;

    given("I am a User attempting to navigate AlarmDelay Screen", () => {
      AlarmDelayWrapper = shallow(<AlarmDelay {...screenProps} />);
      expect(AlarmDelayWrapper).toBeTruthy();
      expect(AlarmDelayWrapper).toMatchSnapshot();
    });

    when("I navigate to the AlarmDelay Screen",async () => {
      instance = AlarmDelayWrapper.instance() as AlarmDelay;

      instance.state = {
            
        Loader: false,
        value1: 10,
        value2: 5,
        Userdata: "",
        Token: "",
        
    };
   

    });

    then("I can open the modal", async () => {
      await instance.setclick()


      instance.setState({ Loader: true })

      const SetsettingID = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      SetsettingID.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        SetsettingID.messageId
      );
      SetsettingID.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        JSON.parse(JSON.stringify(CreateEventRes))
      );
      instance.SetsettingID = SetsettingID.messageId;
      runEngine.sendMessage("Unit Test", SetsettingID);


    let buttonComponent = AlarmDelayWrapper.findWhere((node) => node.prop('testID') === 'setClickId');
    buttonComponent.simulate('press', instance.setclick());
     
    });

    then("I can press call menu", async () => {
      instance.goSignupScreen()
    });

    then("I can press bot menu", async () => {
      const switchbtnComponent = AlarmDelayWrapper.findWhere(
        (node) => node.prop("testID") === "silderValue"
      );
      switchbtnComponent.simulate("valueChange");

      const switchbtnComponent3 = AlarmDelayWrapper.findWhere(
        (node) => node.prop("testID") === "sildervalue2"
      );
      switchbtnComponent3.simulate("valueChange");

      
      const switchbtnComponent1 = AlarmDelayWrapper.findWhere(
        (node) => node.prop("testID") === "SetClickId"
      );
      switchbtnComponent1.simulate("press");

     await instance.setState({ Userdata: "" })
     await instance.setState({ 
        Token: "" })
      instance.setState({
         value1: "", 
         value2: "" 
        })

    },300000);

    then("I can press file menu", async () => {
      instance.setState({Token:""})
      instance.setState({value1:10,value2:5})
     
    });

    then("I can press saved menu", async () => {
     
     
    });

    then("I can press alerts menu", async () => {
     
    
    });

    then("I can press notification menu", async () => {
     
    });

    then("I can press setting menu", async () => {
      
    });
    then('I can leave the screen with out errors', () => {
      instance.componentWillUnmount()
     
      
  });
  then('Handle back button press',() => {
   

    let buttonComponent1 = AlarmDelayWrapper.findWhere((node) => node.prop('testID') === 'backBtnId');
    buttonComponent1.simulate('press');
    instance.goback()

})
  });


});
