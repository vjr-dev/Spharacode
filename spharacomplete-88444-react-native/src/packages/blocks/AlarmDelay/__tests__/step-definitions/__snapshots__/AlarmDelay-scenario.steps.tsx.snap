// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlarmDelay User navigates to AlarmDelay 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/AlarmDelay/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/AlarmDelay/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": 15,
                "justifyContent": "center",
                "padding": 25,
                "width": 15,
              }
            }
            testID="backBtnId"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/AlarmDelay/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "tintColor": "#f07233",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 20,
              "marginTop": 78,
              "width": 730,
            }
          }
        >
          Alarm Delay.
        </Text>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "fontSize": 16,
              "marginTop": 20,
              "width": 730,
            }
          }
        >
          Set number of seconds from when the SOS button is pushed until the alarm triggerd.
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 98,
              "justifyContent": "space-between",
              "marginTop": 25,
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#cdcdcd",
                "fontSize": 20,
                "marginLeft": 20,
              }
            }
          >
            Voice Activation alarm delay.
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
              }
            }
            testID="setClickId"
          >
            <Text
              style={
                Object {
                  "color": "rgba(255,138,73,1)",
                  "fontSize": 20,
                  "marginRight": 29,
                }
              }
            >
              SET
            </Text>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#404040",
              "flexDirection": "row",
              "height": 196,
              "justifyContent": "space-evenly",
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#cdcdcd",
                "marginLeft": 59,
                "marginTop": 31,
              }
            }
          >
            5
          </Text>
          <View
            style={
              Object {
                "height": "100%",
                "justifyContent": "center",
                "width": 628,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(255,211,154,1)",
                  "borderRadius": 98,
                  "left": 41,
                  "width": 196,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0,0,0)",
                    "textAlign": "center",
                  }
                }
              >
                10
                 Second
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginTop": 5,
                  "width": 628,
                }
              }
            >
              <ForwardRef(SliderComponent)
                disabled={false}
                inverted={false}
                maximumTrackTintColor="#a1a1a1"
                maximumValue={60}
                minimumTrackTintColor="rgba(255,138,73,1)"
                minimumValue={5}
                onValueChange={[Function]}
                step={1}
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "width": 539,
                  }
                }
                testID="silderValue"
                thumbTintColor="rgb(255,255,255)"
                value={10}
              />
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#cdcdcd",
                "marginRight": 59,
                "marginTop": 31,
              }
            }
          >
            60
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 98,
              "justifyContent": "space-between",
              "marginTop": 25,
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#cdcdcd",
                "fontSize": 20,
                "marginLeft": 20,
              }
            }
          >
            Gesture trigger delay.
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
              }
            }
            testID="SetClickId"
          >
            <Text
              style={
                Object {
                  "color": "rgba(255,138,73,1)",
                  "fontSize": 20,
                  "marginRight": 29,
                }
              }
            >
              SET
            </Text>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#404040",
              "flexDirection": "row",
              "height": 196,
              "justifyContent": "space-evenly",
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#cdcdcd",
                "marginLeft": 59,
                "marginTop": 31,
              }
            }
          >
            5
          </Text>
          <View
            style={
              Object {
                "height": "100%",
                "justifyContent": "center",
                "width": 628,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(255,211,154,1)",
                  "borderRadius": 98,
                  "left": 1,
                  "width": 196,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(0,0,0)",
                    "textAlign": "center",
                  }
                }
              >
                5
                 Second
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginTop": 5,
                  "width": 628,
                }
              }
            >
              <ForwardRef(SliderComponent)
                disabled={false}
                inverted={false}
                maximumTrackTintColor="#a1a1a1"
                maximumValue={60}
                minimumTrackTintColor="rgba(255,138,73,1)"
                minimumValue={5}
                onValueChange={[Function]}
                step={1}
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "width": 539,
                  }
                }
                testID="sildervalue2"
                thumbTintColor="rgb(255,255,255)"
                value={5}
              />
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#cdcdcd",
                "marginRight": 59,
                "marginTop": 31,
              }
            }
          >
            60
          </Text>
        </View>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 100,
                  "justifyContent": "center",
                  "width": 730,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;
